### PR TITLE
Remove unwanted tags from `wpo_wcpdf_sanitize_html_content()`

### LINKED ISSUE #791: The custom_field() method outputs additional tags
It seems that since version `3.8.1` where we introduced the `wpo_wcpdf_sanitize_html_content()` method, the `custom_field()` is now wrapping the output in a paragraph (`<p>`) and adding a line break at the end of it (`<br>`).

It seems in [this](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/1d26f6d33a64a3413e92f1ca6c9dec47f59f3b7c/includes/wcpdf-functions.php#L585) line, those options added to the `loadHTML()` will cause that additional `<p>` tag. The `DOMDocument` will probably still wrap any standalone text in a `<p>` tag because it tries to enforce a valid HTML structure where each piece of text content must be inside a tag.
One solution would be wrapping the text with a `<div>` tag and removing it later. So we will prevent that auto additional `<p>` tag. 

Also, it seems the `saveHTML()` adds a line break at the end of the content. So, adding a trim can solve the issue.

### PR DESCRIPTION
**Pull request by MohamadNateqi**
close #791

### CODE REVIEW

File: includes/wcpdf-functions.php [hunk 1]
```diff
@@ -583,11 +583,20 @@ function wpo_wcpdf_sanitize_html_content( string $html, string $context = '', ar
 	}
 
 	libxml_use_internal_errors( true ); // suppress malformed HTML errors
-	@$dom->loadHTML( $html, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+	@$dom->loadHTML( '<div>' . $html . '</div>', LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
 	libxml_clear_errors();
 
-	if ( empty( $dom ) ) {
-		return '';
+	$extra_wrapper = $dom->getElementsByTagName( 'div' )->item( 0 );
```
[the remark this answers is not in the document]
Comment (MohamadNateqi) [reply includes/wcpdf-functions.php:589]: I think it should not happen, since I manually added a DIV element, but I'll add a check for that.

**Review by alexmigf — APPROVED** [review]
Works fine on my tests.

**Review by Terminator-Barbapapa — CHANGES_REQUESTED** [review]
It seems this is introducing additional whitespaces where there were none before. Like in the addresses:

Scherm­afbeelding 2024-05-22 om 14 56 34

**Comment by MohamadNateqi** [comment]
@Terminator-Barbapapa 

It's not that this PR is wrapping the text around whitespaces; the reason for that whitespaces, as you showed, is that we have the address field in a new line in our template without any HTML tag wrapping it:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/e2399ebe4ad708447c76da2b5bf7ae2d873a7177/templates/Simple/invoice.php#L55

The line breaks before and after outputting the address cause that. There were no whitespaces if it was immediately after the TD tag in one line.

But, the reason we hadn't this issue before this PR was that the DOM object was automatically wrapping the text around a P tag:

image


With the change of this PR, that P tag no longer exists, so the address is just plain text, and we have extra whitespace because of linebreaks.

We can update all templates and wrap addresses around a tag in another PR.

**Comment by Terminator-Barbapapa** [comment]
@MohamadNateqi Thank you for clarifying. Can you create issues in the correct repositories to update the templates?

**Review by Terminator-Barbapapa — APPROVED** [review]

**Pull request merged**